src/Backend/Modules/Blog/Domain/CommentRepository.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Backend\Modules\Blog\Domain;
+
+use Common\Locale;
+use Doctrine\ORM\EntityRepository;
+
+class CommentRepository extends EntityRepository
+{
+    public function listCountPerStatus(Locale $locale) :array
+    {
+        $builder = $this->getEntityManager()
+                        ->createQueryBuilder()
+                        ->select('c.status, count(c.status) as number')
+                        ->from(Comment::class, 'c')
+                        ->where("c.locale = :locale")
+                        ->setParameter('locale', $locale)
+                        ->groupBy('c.status');
+
+        $results = $builder->getQuery()->getResult();
+        $data = [];
+
+        foreach ($results as $row) {
```

Comment:
you can do this with a fairly simple `return array_reduce(...);`

Reply:
Hmm, no.

Reply:
Yeah I misread what you are actually doing. A foreach loop is fine